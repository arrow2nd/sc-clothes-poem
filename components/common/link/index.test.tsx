import { render } from '@testing-library/react'

import Link from './index'

describe('Link', () => {
  test.each`
    className    | children
    ${'m-1'}     | ${(<span>TEST</span>)}
    ${'m-1'}     | ${undefined}
    ${undefined} | ${(<p>test</p>)}
    ${undefined} | ${undefined}
  `('設定したPropsが反映されているかどうか', (props) => {
    const { container } = render(
      <Link {...props} title="test" href="http://example.com/" />
    )
    expect(container).toMatchSnapshot()
  })
})
