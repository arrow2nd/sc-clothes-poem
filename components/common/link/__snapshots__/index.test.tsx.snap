// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link 設定したPropsが反映されているかどうか 1`] = `
<div>
  <a
    class="m-1"
    href="http://example.com/"
    rel="noopener noreferrer"
    target="_blank"
    title="test"
  >
    <span>
      TEST
    </span>
  </a>
</div>
`;

exports[`Link 設定したPropsが反映されているかどうか 2`] = `
<div>
  <a
    class="m-1"
    href="http://example.com/"
    rel="noopener noreferrer"
    target="_blank"
    title="test"
  >
    test
  </a>
</div>
`;

exports[`Link 設定したPropsが反映されているかどうか 3`] = `
<div>
  <a
    href="http://example.com/"
    rel="noopener noreferrer"
    target="_blank"
    title="test"
  >
    <p>
      test
    </p>
  </a>
</div>
`;

exports[`Link 設定したPropsが反映されているかどうか 4`] = `
<div>
  <a
    href="http://example.com/"
    rel="noopener noreferrer"
    target="_blank"
    title="test"
  >
    test
  </a>
</div>
`;
